src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/ArcParser.java

```diff
@@ -238,6 +240,34 @@ private static void loadArcHeaderNotZipped(RandomAccessFile raf, ArcEntry arcEnt
   }
   
 
+  public static BufferedInputStream lazyLoadBinary(String arcFilePath, long arcEntryPosition) throws Exception{
+      ArcEntry arcEntry = new ArcEntry(); // We just throw away the header info anyway 
+      
+      if (arcFilePath.endsWith(".gz")){ //It is zipped
+          
+          RandomAccessFile raf = new RandomAccessFile(new File(arcFilePath), "r");
+          raf.seek(arcEntryPosition);          
+
+          // log.info("file is zipped:"+arcFilePath);
+          InputStream is = Channels.newInputStream(raf.getChannel());                           
+          GZIPInputStream zipStream = new GZIPInputStream(is);              
+          BufferedInputStream  bis= new BufferedInputStream(zipStream);
+
+          loadArcHeaderZipped(bis, arcEntry);
```

Comment:
I cannot figure out why there are different parsers for zipped and non-zipped headers. They both operate on an unzipped `InputStream`!?

Reply:
HHMM. Good question, not sure it is correct. But this must be another refactorng PR. It was not part of this PR.